### PR TITLE
feat: selectSignal config

### PR DESCRIPTION
**Pull request by timdeschryver**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #

## What is the new behavior?

This PR adds the `ValueEqualityFn` configuration to the `selectSignal` API in `@ngrx/store` and `@ngrx/component-store`.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: modules/component-store/src/component-store.ts [hunk 1]
```diff
@@ -34,6 +34,7 @@ import {
   isDevMode,
   Signal,
   computed,
+  ValueEqualityFn,
```
Comment (brandonroberts) [inline modules/component-store/src/component-store.ts:37]: Should this be imported as a `type`?

Comment (timdeschryver) [reply modules/component-store/src/component-store.ts:37]: Yes, that would be better. Thanks!

**Review by brandonroberts — APPROVED** [review]

File: modules/component-store/src/component-store.ts [hunk 2]
```diff
@@ -42,6 +43,13 @@ export interface SelectConfig {
   debounce?: boolean;
 }
 
+export interface CreateSelectComponentSignalOptions<T> {
```
[the remark this answers is not in the document]
Comment (timdeschryver) [reply modules/component-store/src/component-store.ts:46]: Yea, I didn't come up with a better name so I resorted to the Angular naming.
`SelectSignalOptions` is better though, thanks!

File: modules/store/src/models.ts [hunk 3]
```diff
@@ -169,3 +171,10 @@ export interface RuntimeChecks {
    */
   strictActionTypeUniqueness?: boolean;
 }
+
+export interface CreateSelectSignalOptions<T> {
```
[the remark this answers is not in the document]
Comment (timdeschryver) [reply modules/store/src/models.ts]: Should we come up with different names for component store and store?

[the remark this answers is not in the document]
Comment (markostanimirovic) [reply modules/store/src/models.ts]: I think it's fine to name them the same until they're private APIs in both packages.

**Review by markostanimirovic — APPROVED** [review]
🚀

**Pull request merged**